package j10_exception;

// ** Exception

// 1) Exception 처리하지 않는 경우
// => Exception 발생 위치에서 비정상 종료

// 2) Exception 처리하는 경우
// => Exception 발생 시 대응을 하여 정상적인 진행

public class Ex01_Basic {

	public static void main(String[] args) {
		int x = 6, y = 3, result = 0;
		String s = "123000";
		String[] names = { "Apple", "Banana", "Coffee" };

		// Test 1 : Exception 처리하지 않는 경우 -> 비정상 종료
		y = 0;
		result = x / y;
		// System.out.println("** result 1 : " + result);
		// => Exception in thread "main" java.lang.ArithmeticException: / by zero

//		==================================

		// Test 2 : Exception 처리하는 경우
		y = 3;

		try {

		} catch (Exception e) {

		} finally {

		}

	}// main

}// class